### PR TITLE
Raise warning for FirstLevelModel fit when confounds and design_matrices are both specified

### PR DESCRIPTION
**Pull request by achamma723**
Changes proposed in this pull request:
- Adding a warning in the fit function when both the design matrices and (confounds and events) are provided in #2704

### CODE REVIEW

File: nilearn/glm/first_level/first_level.py
Comment (NicolasGensollen) [inline nilearn/glm/first_level/first_level.py:522]: I would expect it to be given at the beginning of the function rather than at the end, but otherwise the warning looks good.

**Review by NicolasGensollen — CHANGES_REQUESTED** [review]
Thanks for addressing this @achamma723 !
Could you add a test checking that the warning is given correctly when providing a design matrix and confounds?

**Comment by NicolasGensollen** [comment]
Linking #2704

**Comment by achamma723** [comment]
@NicolasGensollen Can you tell more about this error?

File: nilearn/glm/tests/test_first_level.py
Comment (NicolasGensollen) [inline nilearn/glm/tests/test_first_level.py:635]: I think this is too complicated for a simple warning check. I would be in favor of just checking this in `test_fmri_inputs` for example by adding something like:

```python
with pytest.warns(UserWarning,
                  match="If the design matrices are given"):
        FirstLevelModel(mask_img=mask).fit(fi, 
                                           design_matrices=d,
                                           confounds=conf)
```
WDYT?

**Review by NicolasGensollen — CHANGES_REQUESTED** [review]
I think we can simplify the test a lot (see my comment below). 
I haven't checked the Azure error you're getting yet, but I think we can update the test before debugging.

**Comment by achamma723** [comment]
@NicolasGensollen Everything is good now. Thanks for your remarks!

File: nilearn/glm/first_level/first_level.py
Comment (tsalo) [inline nilearn/glm/first_level/first_level.py:367]:
```suggestion
            warn('If design matrices are given, they take precedence over events and confounds.')
```

I'd lean toward using simpler words though. Perhaps `"If design matrices are supplied, confounds and events will be ignored."`.

File: nilearn/glm/first_level/first_level.py
Comment (tsalo) [inline nilearn/glm/first_level/first_level.py:366]:
```suggestion
        if design_matrices is not None and (confounds is not None or events is not None):
```
Based on the warning, it seems like both confounds and events should be checked here.

**Review by tsalo — CHANGES_REQUESTED** [review]
Things look great. I just have a couple of minor requests. Additionally, would you mind (1) simplifying the PR title and (2) linking directly to #2704 with a [closing keyword](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (e.g., "Closes ")?

**Review by tsalo — APPROVED** [review]
The changes look good to me. Will need to wait for @NicolasGensollen to re-review before merging though.

**Review by NicolasGensollen — APPROVED** [review]
This looks good to me too! :tada: 
Thanks for working on this @achamma723 and thanks for the reviews @tsalo ! :+1:

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by NicolasGensollen** [comment]
Thanks everybody! Merging...

**Pull request merged**
